### PR TITLE
deps: replace "nyc" with "c8"

### PR DESCRIPTION
**Pull request by justinmk3**
## Problem

`nyc` is no longer maintained: https://github.com/istanbuljs/nyc/issues/1514

and it depends on various outdated packages, .e.g `uuid@3.4.0` (check with `npm ls uuid`).

## Solution

- Remove `nyc`.
- Migrate to `c8`.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinmk3** [comment]
Looks like it worked: https://app.codecov.io/github/aws/aws-toolkit-vscode/commit/bc308d44c8cca822ed4a982195175c8adc6b70fa

```
/Users/runner/work/_actions/codecov/codecov-action/v3/dist/codecov -n  -Q github-action-3.1.4 -f ./coverage/lcov.info -F macos-unittests -C bc308d44c8cca822ed4a982195175c8adc6b70fa -v
[2024-01-04T21:09:34.038Z] ['verbose'] Start of uploader: 1704402574038...
[2024-01-04T21:09:34.040Z] ['info'] 
     _____          _
    / ____|        | |
   | |     ___   __| | ___  ___ _____   __
   | |    / _ \ / _` |/ _ \/ __/ _ \ \ / /
   | |___| (_) | (_| |  __/ (_| (_) \ V /
    \_____\___/ \__,_|\___|\___\___/ \_/
  Codecov report uploader 0.7.1
[2024-01-04T21:09:34.065Z] ['info'] => Project root located at: /Users/runner/work/aws-toolkit-vscode/aws-toolkit-vscode
[2024-01-04T21:09:34.076Z] ['info'] -> No token specified or token is empty
[2024-01-04T21:09:34.076Z] ['verbose'] Start of network processing...
[2024-01-04T21:09:34.076Z] ['verbose'] Searching for files in /Users/runner/work/aws-toolkit-vscode/aws-toolkit-vscode
[2024-01-04T21:09:34.1[57](https://github.com/aws/aws-toolkit-vscode/actions/runs/7414756522/job/20176622921?pr=4228#step:7:58)Z] ['verbose'] coveragepy is not installed
[2024-01-04T21:09:34.157Z] ['info'] Searching for coverage files...
[2024-01-04T21:09:35.034Z] ['verbose'] Preparing to clean the following coverage paths: ./coverage/lcov.info
[2024-01-04T21:09:35.034Z] ['info'] => Found 1 possible coverage files:
  ./coverage/lcov.info
[2024-01-04T21:09:35.034Z] ['verbose'] End of network processing
[2024-01-04T21:09:35.034Z] ['info'] Processing ./coverage/lcov.info...
```

File: .c8rc.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "report-dir": "./coverage",
+    "reporter": ["lcov"],
+    "all": true,
+    "include": ["src/**", "dist/src/**"],
```
Comment (hayemaxi) [inline .c8rc.json:5]: should we check the lint rules in plugins/ as well?

**Review by hayemaxi — APPROVED** [review]

File: .c8rc.json (hunk 1 above)
Comment (justinmk3) [reply .c8rc.json:5]: Sure, let's add that later.

File: .c8rc.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    "exclude": [
+        "dist/src/ssmDocument/ssm/ssmServer.js",
```
Comment (justinmk3) [inline .c8rc.json:7]: Weird failure from somewhere in `ssmDocument/**` : 

```
Error: ENOENT: no such file or directory, open 'D:\a\aws-toolkit-vscode\aws-toolkit-vscode\coverage\lcov-report\dist\src\ssmDocument\external "buffer".html'
    at Object.openSync (node:fs:590:3)
    at FileWriter.writeFile (D:\a\aws-toolkit-vscode\aws-toolkit-vscode\node_modules\istanbul-lib-report\lib\file-writer.js:185:41)
```

**Pull request merged**